### ISSUE TITLE
LT08 vs LT01 conflict in recursive CTE with cycle detection

### ISSUE THREAD

**Issue by jasonaowen** [sqlfluff/sqlfluff#7214]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Using the default rules, linting a [recursive common table expression with cycle detection](https://www.postgresql.org/docs/current/queries-with.html#QUERIES-WITH-CYCLE) always fails due to conflicting rules.

`LT01` requires no line break between the closing paren and the `CYCLE` keyword, but `LT08` requires a newline after the closing paren.

Running `fix` repeatedly produces three different layouts in a loop, each of which fail a subsequent `lint`.

### Expected Behaviour

I would expect there to exist some layout that would satisfy the default rules, and that running `fix` would produce that layout.

### Observed Behaviour

```sh
$ sqlfluff fix --dialect postgres test.sql
==== finding fixable violations ====
WARNING    One fix for LT01 not applied, it would re-cause the same error.
WARNING    One fix for LT01 not applied, it would re-cause the same error.
== [test.sql] FAIL
L:   6 | P:   3 | LT08 | Blank line expected but not found after CTE closing
                       | bracket. [layout.cte_newline]
== [test.sql] FIXED
1 fixable linting violations found

$ sqlfluff lint --dialect postgres test.sql 
== [test.sql] FAIL                                                                                                            
L:   6 | P:   2 | LT01 | Unexpected line break. [layout.spacing]
L:   7 | P:   1 | LT01 | Unexpected line break. [layout.spacing]
L:   8 | P:   1 | LT01 | Expected single whitespace between end bracket ')' and
                       | 'CYCLE' keyword. [layout.spacing]
All Finished 📜 🎉!
```

### How to reproduce

```sh
sqlfluff lint --dialect postgres test.sql
```

`test.sql`:
```sql
WITH RECURSIVE t (n) AS (
    VALUES (1)
    UNION ALL
    SELECT n + 1 FROM t
    WHERE n < 10
) CYCLE n SET is_cycle USING path

SELECT * FROM t;
```

### Dialect

PostgreSQL

### Version

- Python 3.12.3
- sqlfluff, version 3.5.0

### Configuration

Default rules

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Fixed by #7468
